packages/sign-client/test/sdk/client.spec.ts

```diff
@@ -112,36 +112,62 @@ describe("Sign Client Integration", () => {
           deleteClients(clients);
         });
       });
-      it("clients can ping each other after restart", async () => {
-        const beforeClients = await initTwoClients(
-          {
-            storageOptions: { database: TEST_SIGN_CLIENT_A_DATABASE },
-          },
-          {
-            storageOptions: { database: TEST_SIGN_CLIENT_B_DATABASE },
-          },
-        );
-        const {
-          pairingA: { topic },
-        } = await testConnectMethod(beforeClients);
-        // ping
-        await beforeClients.A.ping({ topic });
-        await beforeClients.B.ping({ topic });
-        // delete
-        deleteClients(beforeClients);
-        // restart
-        const afterClients = await initTwoClients(
-          {
-            storageOptions: { database: TEST_SIGN_CLIENT_A_DATABASE },
-          },
-          {
-            storageOptions: { database: TEST_SIGN_CLIENT_B_DATABASE },
-          },
-        );
-        // ping
-        await afterClients.A.ping({ topic });
-        await afterClients.B.ping({ topic });
-        deleteClients(afterClients);
+      describe("after restart", () => {
+        let beforeClients;
+        let afterClients;
+        beforeEach(async () => {
+          beforeClients = await initTwoClients(
+            {
+              storageOptions: { database: TEST_SIGN_CLIENT_A_DATABASE },
+            },
+            {
+              storageOptions: { database: TEST_SIGN_CLIENT_B_DATABASE },
+            },
+          );
+        });
+        afterEach(async (done) => {
+          const { result } = done.meta;
+          if (result?.state.toString() !== "pass") {
+            if (!beforeClients) {
+              console.log('Clients failed to initialize');
+              return;
+            }
+            console.log(
+              `Test ${
+                done.meta.name
+              } failed with before client ids: A:'${await beforeClients.A.core.crypto.getClientId()}';B:'${await beforeClients.B.core.crypto.getClientId()}'`,
+            );
```

Comment:
```
FAIL  test/sdk/client.spec.ts > Sign Client Integration > ping > pairing > after restart > clients can ping each other
[132](https://github.com/WalletConnect/rs-relay/actions/runs/3088688370/jobs/4997608673#step:4:133)
   ❯ test/sdk/push.spec.ts > Push > receives a prompt webhook
[133](https://github.com/WalletConnect/rs-relay/actions/runs/3088688370/jobs/4997608673#step:4:134)
     → Cannot read properties of undefined (reading 'topic')
[134](https://github.com/WalletConnect/rs-relay/actions/runs/3088688370/jobs/4997608673#step:4:135)

[135](https://github.com/WalletConnect/rs-relay/actions/runs/3088688370/jobs/4997608673#step:4:136)
TypeError: Cannot read properties of undefined (reading 'core')
[136](https://github.com/WalletConnect/rs-relay/actions/runs/3088688370/jobs/4997608673#step:4:137)
 ❯ test/sdk/client.spec.ts:138:74
[137](https://github.com/WalletConnect/rs-relay/actions/runs/3088688370/jobs/4997608673#step:4:138)
    136|               `Test ${
[138](https://github.com/WalletConnect/rs-relay/actions/runs/3088688370/jobs/4997608673#step:4:139)
    137|                 done.meta.name
[139](https://github.com/WalletConnect/rs-relay/actions/runs/3088688370/jobs/4997608673#step:4:140)
    138|               } failed with before client ids: A:'${await beforeClient…
[140](https://github.com/WalletConnect/rs-relay/actions/runs/3088688370/jobs/4997608673#step:4:141)
       |                                                                          ^
[141](https://github.com/WalletConnect/rs-relay/actions/runs/3088688370/jobs/4997608673#step:4:142)
    139|             );
[142](https://github.com/WalletConnect/rs-relay/actions/runs/3088688370/jobs/4997608673#step:4:143)
    140|             if (!afterClients) return;
[143](https://github.com/WalletConnect/rs-relay/actions/runs/3088688370/jobs/4997608673#step:4:144)
⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/3]⎯
```

https://github.com/WalletConnect/rs-relay/actions/runs/3088688370/jobs/4997608673

I have no idea under what condition this happens. Any ideas @bkrem ?

Reply:
Seems like the client(s) didn’t initialise properly if eg `A.core` isn’t available, since it’s throwing on the attempted access to the clientId via the core module